src/Cameras/VR/babylon.vrExperienceHelper.ts

```diff
@@ -44,6 +44,10 @@ module BABYLON {
          */
         public onEnteringVRObservable = new Observable<VRExperienceHelper>();
 
+         /**
+         * Observable raised when VR Display is ready.
+         */
+        public onVRDeviceReadyObservable = new Observable<VRExperienceHelper>();

```

Comment:
maybe onVRDisplayChangedObservable? Ready might be a bit misleading here.

Reply:
Sure @RaananW , on it.

Reply:
@RaananW , VRExperienceHelper uses this._webVRready to consider it ready to go for VR. If I rename it to onVRDisplayChangedObservable I should not need to check _webVRReady is true I guess...?

Reply:
Maybe changing the location of the observable notification?

 if (webVROptions.createFallbackVRDeviceOrientationFreeCamera) {
                this.displayVRButton();
            } else {
                this._scene.getEngine().onVRDisplayChangedObservable.add((e) => {
                    
                    this.onVRDisplayChangedObservable.notifyObservers(e);

                    if (e.vrDisplay) {
                        this.displayVRButton();
                    }
                })
            }

Reply:
You mean in the constructor? looks fine.

As I see, it seems like prepending a function to the engine.onVRDisplayChangedObservable will be enough in your case won't it?